### DISCUSSION TITLE
`context` segment for switching user with p10k

### DISCUSSION THREAD

**Discussion by Syphdias** [romkatv/powerlevel10k#2699] [General]
I sometimes `sudo -iu user` or `sudo su - user` to user `user`. And in these cases it would be nice to have the context menu or at least the username permanently in the prompt to be able to differentiate between terminal windows.

Does anyone know if I can detect if I switched user in either or both these cases, so I can enable the context segment?

I don't think `sudo -E` would be the right fit for two reasons. 1. I didn't get it to work with non-root 2. The user's p10k should be used.

**Comment by romkatv** [comment]
`sudo -iu user` defines a bunch of `SUDO_*` environment variables, so you could look at those.

To detect `sudo su - user`, you could look at the output of `ps -o cmd= -p $PPID`.

**Reply by Syphdias** [reply]
Thank you! I also found something in the `.p10k.zsh` solving my problem. I believe this already takes `SUDO_COMMAND` into account.

There is this section:
```zsh
  # Don't show context unless running with privileges or in SSH.
  # Tip: Remove the next line to always show context.
  typeset -g POWERLEVEL9K_CONTEXT_{DEFAULT,SUDO}_{CONTENT,VISUAL_IDENTIFIER}_EXPANSION=
```
But since I don't want to show it always I only removed the `SUDO` variables:
 ```zsh
   typeset -g POWERLEVEL9K_CONTEXT_DEFAULT_{CONTENT,VISUAL_IDENTIFIER}_EXPANSION=
 ```
 I guess this overwrites the the default expansion of `%n@%m` to nothing for `DEFAULT` but not for `SUDO` .
